students/daniel_grubbs/lesson05/mailroom3.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python3
+
+# Week 5
+# Mailroom Assignment - Part 2
+#
+# Changelog:
+# - Added in the use of dictionaries
+# - Switched to using switch-case
+# - Added in try-except block
+# - Changed print_donor_list() to a list comprehension
+
+import sys
+
+donors = {
+    'Jimmy Nguyen': [100, 1350, 55],
+    'Steve Smith': [213, 550, 435],
+    'Julia Norton': [1500, 1500, 1500],
+    'Ed Johnson': [150],
+    'Elizabeth McBath': [10000, 1200]
+}
+
+
+def print_donor_list():
+    """Function simply for looping through the donor list.
+    Made into function since it may be called multiple times.
+    """
+    # for donor in donors.keys():
+    #     print(donor)
+
+    donor = [name for name in donors.keys()]
+    return donor
+
+
+def get_donor(name):
+    """Get donor from the donors dictionary."""
+    donor = name.lower()
+    for k in donors.keys():
+        if donor == k.strip().lower():
+            return k
+        else:
+            return None
+
+    # donor = [k for k in donors.keys() if name == k.strip().lower()]
```

Comment:
lol.  I was trying to do this same thing in your mailroom2 before i scrolled here but couldn't get the `else None` to work, even though i think it should.  I think this would be a good one to bring up to Rick for office hours.